datatypes/src/collections/feature_collection_builder.rs

```diff
@@ -166,19 +166,11 @@ where
         };
 
         // check that data types match
-        // TODO: think of cheaper call for checking data type match
-        let data_type_variant = mem::discriminant(&FeatureDataType::from(&data));
-        match self.types.get(column) {
-            Some(data_type) if data_type_variant != mem::discriminant(data_type) => {
-                return Err(FeatureCollectionError::WrongDataType.into());
-            }
-            None => {
-                return Err(FeatureCollectionError::ColumnDoesNotExist {
-                    name: column.to_string(),
-                }
-                .into());
-            }
-            Some(_) => (),
+        let expected_type = mem::discriminant(self.types.get(column).expect("checked before"));
```

Comment:
It isn't checked before unfortunately.

You could say something like this
```Rust
        let (data_builder, column_type) = if let (Some(builder), Some(column_type)) = (self.builders.get_mut(column), self.types.get(column)) {
            (builder, column_type)
        } else {
            return Err(FeatureCollectionError::ColumnDoesNotExist {
                name: column.to_string(),
            }
            .into());
        };
```

Reply:
I was inspired by the `push_null` function below. It looks quite the same and also uses this trick (inside the match expression).
Can't we assume that if a builder exists for a given column then a type must be set in `types`?

Reply:
I guess we could. But since we are already restructuring, we could add it there.
You can see it that way: One potential panic less in the code :smile: without real overhead.

Reply:
Yeah, the overhead is basically non-existent 😂 After I changed the code I realised that we can omit the call to mem::discriminant and compare the FeatureDataType-values directly.

4fd1494